sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+        "ModelDataUrl": model_url_s3,
+        "Environment": {"SAGEMAKER_PROGRAM": entrypoint_file_name}
+    }
+
+    sm_model_name = model_name + '-' + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+    create_model_response = sm_client.create_model(ModelName=sm_model_name,
+                                                   ExecutionRoleArn=role, Containers=[container])
+    logging.info('Model ARN: {}'.format(create_model_response['ModelArn']))
+
+    # ================= create an endpoint configuration ================= #
+    sm_ep_config_name = model_name + "-EndpointConfig-" + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+    create_endpoint_config_response = sm_client.create_endpoint_config(
+            EndpointConfigName=sm_ep_config_name,
+            ProductionVariants=[
+                {
+                    "InstanceType": input_args.instance_type,
+                    "InitialInstanceCount": input_args.instance_count,
+                    "InitialVariantWeight": 1,
+                    "ModelName": sm_model_name,
+                    "VariantName": "AllTraffic",
+                }
+            ],
+        )
+
+    logging.info("Endpoint config Arn: {}".format(create_endpoint_config_response["EndpointConfigArn"]))
+
+    # ================= create an endpoint ================= #
+    sm_ep_name = model_name + "-Endpoint-" + strftime("%Y-%m-%d-%H-%M-%S", gmtime())
+
+    create_endpoint_response = sm_client.create_endpoint(
+        EndpointName=sm_ep_name, EndpointConfigName=sm_ep_config_name
+    )
+    logging.info("Endpoint Arn: {}".format(create_endpoint_response["EndpointArn"]))
+
+    resp = sm_client.describe_endpoint(EndpointName=sm_ep_name)
+    status = resp["EndpointStatus"]
+    logging.info("Endpoint Status: " + status)
+
+    logging.info("Waiting for {} endpoint to be in service...".format(sm_ep_name))
+    waiter = sm_client.get_waiter("endpoint_in_service")
+    
+    try:
+        waiter.wait(EndpointName=sm_ep_name,
+                    WaiterConfig={
+                        'Delay': 30,        # seconds between querying
+                        'MaxAttempts': 60   # max retries (~30 minutes here)
+                    })
+        logging.info('{} endpoint has been successfully created, and ready to be \
+                 invoked!'.format(sm_ep_name))
+    except WaiterError as e:
+        logging.error("Waiter timed out or endpoint creation failed:", e)
+
+
+def get_realtime_infer_parser():
+    """
+    Get GraphStorm realtime  task parser.
+    """
+    realtime_infer_parser = argparse.ArgumentParser("GraphStorm Inference Args")
+
+    # SageMaker specific arguments
+    realtime_infer_parser.add_argument("--image-url", type=str, required=True,
+        help="GraphStorm SageMaker docker image URI")
+    realtime_infer_parser.add_argument("--role", type=str, required=True,
+        help="SageMaker execution role")
+    realtime_infer_parser.add_argument("--region", type=str, required=True,
```

Comment:
Shall we add some check for this region is where the inference image and model tar file are located?

Reply:
Yes. added a simple check if the docker image is in the same region